llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -3903,9 +3919,14 @@ FixedScalableVFPair LoopVectorizationCostModel::computeFeasibleMaxVF(
   // dependence distance).
   unsigned MaxSafeElements =
       llvm::bit_floor(Legal->getMaxSafeVectorWidthInBits() / WidestType);
+  unsigned MaxScalableSafeElements = MaxSafeElements;
+  if (foldTailWithEVL() && !Legal->isSafeForAnyVectorWidth()) {
+    MaxScalableSafeElements = std::numeric_limits<unsigned>::max();
```

Comment:
Might be good to document this. IIUC this means if we have something like `MaxSafeElements == 4`, we will still try VFs > 4, even thought we know only at most 4 lanes will execute? Is this desirable? 

Or should this limit to the next power-of-2 of MaxSafeElements?

Reply:
I thought about this and at first I thought it might be good to use max here. But after some thoughts, I think next-power-of-2 is good here

Reply:
Where should I document this?